_high_availability/002-Replication-Configuration.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: page
+title:  "Replication configuration"
+date:   2015-01-27 22:02:36
+categories:
+permalink: /ha/replication.html
+---
+
+
+The master configuration
+=======================
+
+The master server is responsible for distributing the changes to the stand by servers. We have to change to following settings to achieve that:
+
+    port = 5234 # it's a good idea to change the port, we will use a connection pool 
+    listen_addresses = '*' # we are using a star (bind to everything) as we want to reuse the same postgresql.conf to all our nodes
+    wal_level = hot_standy
+    max_wal_senders = 5 # this is the total of wal senders that can be used concurrently by the standbys or streaming base backups
+    wal_keep_segments = 8 # how many wal segments should we keep? This has a relation with the speed with which the standbys consume the logs. Increase if you have slow standbys
+    arhive_mode = on # We keep archives in case we need them for slaves that have fallen behind
+    hot_standby = on # this is ignored by the master server
+
+It is much easier to keep a single postgresql.conf and share it between all your nodes. 
```

Comment:
It is mostly a comment that it is easier to keep a single postgresql.conf in such a setup. 

Given that the servers should be equivalent, in the sense that failover will promote one of them to assume active duties, there shouldn't be serious divergence, otherwise you may end up with an underpowered slave serving live traffic.